### DISCUSSION TITLE
Start / stop Cron from a micro

### DISCUSSION THREAD

**Discussion by xu2xulim** [deta/cloud-docs#275] [Help] [answered]
Is there a way to start / stop a cron from which the micro  eg endpoint receives a trigger? The cron is part of the same micro.

**Comment by abdelhai** [comment, marked as answer]
deta supports setting and "unsetting" a cron from the cli: https://docs.deta.sh/docs/micros/cron#set-cron
you can however write your own logic which could depend on some data/flag stored in base which you could update via the micro endpoint.

this works because cron works with other frameworks: https://docs.deta.sh/docs/micros/cron#cron-and-http

**Reply by xu2xulim** [reply]
I am familiar with the above and had been using them. What I am not familiar is whether I can have an endpoint /{option}/cron and within it I can start and stop a cron. I tried using subprocess but I have not gotten it to work. Perhaps some python expert can offer some directions. TIA

**Reply by abdelhai** [reply]
cron doesn't run on the vm, it's an external service that invokes the vm at your defined intervals. the only way to control the cron behavior is by using the deta cli. you could upload the deta cli with your micro code and invoke it via a shell subprocess there to do what you want. make sure to use an access token https://docs.deta.sh/docs/cli/auth#deta-access-tokens

**Reply by xu2xulim** [reply]
Not sure I understand or know how to do it. I will go ahead a marked it as answered. I am currently running ok using the approach similar to what @jnsougata had suggested.

**Reply by abdelhai** [reply]
essentially what I'm saying is that the only way to control "deta cron" is via our cli.
otherwise, @jnsougata suggestion on how to augment is it seems like a good solution as well.

**Comment by jnsougata** [comment]
Step 1: Store some flags inside base. You can develop some endpoints to update those flags inside the base. 
Step 2: Write some logic inside corn function  that depends on those stored flags.

Let's say you have some key `start` in base with a value `True`
so you can fetch this `True` value form base inside the corn and satisfy some bool logic.
```python
@app.lib.cron()
def cron_job(event):
    start = base.get('start')
    if start:
        # do stuff here.....
        return "Hello Deta, I am a cron job"
```
Now you can make some endpoint to interact with the base and change the value of key `start` from `True` to `False`. Then the code after `if statement` won't run. 

Note: If I were to say to do something like you said I wouldn't have even used corn!

**Reply by xu2xulim** [reply]
Thanks, I understand the logic. I have something like that running to see whether there are accepted XERO quotations to be processed. The cron is set to 1 minute....I have hoping not to have cron running in the background when it has been determined that no records to be processed and to trigger the 1 minute cycle when the next batch of records start to come in.

**Reply by jnsougata** [reply]
I guess what you wan't is dynamic scheduling of jobs!
